cpp/CMakeLists.txt

```diff
@@ -139,6 +139,17 @@ option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 option(BUILD_TESTS "Configure CMake to build tests" ON)
 option(BUILD_BENCHMARKS "Configure CMake to build (google) benchmarks" OFF)
 
+###################################################################################################
+# - cudart options --------------------------------------------------------------------------------
+# cudart can be statically linked or dynamically linked. The python ecosystem wants dynamic linking
+
+if(CUDA_RUNTIME_LIBRARY AND CUDA_RUNTIME_LIBRARY MATCHES "^Static$")
```

Comment:
Can we just make this an `ON` / `OFF` instead of a string match? I.e.:
```
option(CUDA_STATIC_RUNTIME "Statically link the CUDA runtime" OFF)
```

Reply:
Could you make a similar comment on https://github.com/rapidsai/rmm/pull/343 I would like to keep the RMM code and this code in sync so both features behave the same way.

Reply:
I see why you made this use string matching now to try to follow the new CMake 3.17 features that bring native support for this for `.cu` file compilation, but doesn't do it for our `.cpp` files that use the runtime API.